g1ant-language/refining-your-code.md

```diff
@@ -0,0 +1,94 @@
+## Refining Your Code
+
+The code is never good enough. Of course, the golden rule says that if it works, don’t fix it — and we couldn’t agree more. But when you are practicing with programming, refinements are the part of the learning process. You learn new things and you can look back at your previous works to see if it’s possible to use your latest knowledge to optimize the code, make it clearer and more efficient.
```

Comment:
golden rule not to refactor your code? I disagree

Reply:
GENERAL golden rule is: "if it ain't broke, don't fix it"

Reply:
I don't think developers would agree with you, refactor is an important part of developing and without it "if it ain't broke, it most probably will be broke" it's because if your code works but no one knows why, it creates a situation in which only you can work on it and after a while you will find it extremely hard to do. Also, even a small change can do much harm... So it might be a GENERAL rule but not in programming :D

Reply:
Alright, I will make this sentence more precise in its generality :D
"Of course, the golden rule of life says that if it works, don’t fix it — and we couldn’t agree more."

Reply:
hmm that wasn't what I was talking about hahahha I meant that it can remain the same if only you remove "and we couldn't agree more" oh come on! who agrees with that sentence? 😛

Reply:
```suggestion
The code is never good enough. Of course, the golden rule of life says that if it isn't broken, don’t fix it — and we couldn’t agree more. But when you are practicing with programming, refinements are the part of the learning process. You learn new things and you can look back at your previous works to see if it’s possible to use your latest knowledge to optimize the code, make it clearer and more efficient.
```